src/audio/kpb.c

```diff
@@ -373,15 +387,27 @@ static void kpb_buffer_data(struct comp_data *kpb, struct comp_buffer *source)
 		/* We need to split copying into two parts
 		 * and wrap buffer pointer
 		 */
-		memcpy(hb->w_ptr, source->r_ptr, space_avail);
+		if (memcpy_s(hb->w_ptr, space_avail,
+		    source->r_ptr, space_avail)) {
+			trace_kpb_error("kpb_buffer_data()"
+					"unable to copy data");
+		}
 		size_to_copy = size_to_copy - space_avail;
 		hb->w_ptr = hb->sta_addr;
-		memcpy(hb->w_ptr, source->r_ptr, size_to_copy);
+		if (memcpy_s(hb->w_ptr, source->r_ptr,
+		    size_to_copy)) {
+			trace_kpb_error("kpb_buffer_data()"
+					"unable to copy data");
+		}
 		hb->w_ptr += size_to_copy;
 		size_to_copy = 0;
 
 	} else {
-		memcpy(kpb->w_ptr, source->data_ptr, size_to_copy);
+		if (memcpy(kpb->w_ptr, space_avail,
```

Comment:
Probably typo `memcpy` instead of `memcpy_s`

Reply:
After offline conversation with @mrajwa  will drop KPB changes since KPB will be heavily refactored.